smart-contracts/contracts/mixins/MixinGrantKeys.sol

```diff
@@ -25,10 +27,14 @@ contract MixinGrantKeys is
     address[] calldata _keyManagers
   ) external {
     _lockIsUpToDate();
-    require(isKeyGranter(msg.sender) || isLockManager(msg.sender), 'ONLY_LOCK_MANAGER_OR_KEY_GRANTER');
+    if(!isKeyGranter(msg.sender) && !isLockManager(msg.sender)) {
+      revert ONLY_LOCK_MANAGER_OR_KEY_GRANTER();
+    }
 
     for(uint i = 0; i < _recipients.length; i++) {
-      require(_recipients[i] != address(0), 'INVALID_ADDRESS');
+      if(_recipients[i] == address(0)) { 
```

Comment:
The address 0 check can be useful because without it, it will just fails silently

The best is to move that check in `_createNewKey` which is an internal function used by several places (purchase, etc).